File: C/beginner/1072.c
// https://judge.beecrowd.com/pt/problems/view/1072
// Intervalo 2

// Libraries
#include <stdio.h>

// Main Function
int main() {
    // Starting code and declarations
    int usr_input, usr_input2;
    int interval_in, interval_out;

    interval_in = 0;
    interval_out = 0;    

    // Scanning user input
    scanf("%d", &usr_input);

    // Iterating
    for (int i = 1; i <= usr_input; i++) {
        scanf("%d", &usr_input2);
        if (usr_input2 >= 10 && usr_input2 <= 20) {            
            interval_in ++;
        } else {
            interval_out ++;
        }
    }

    // Showing results
    printf("%d in\n", interval_in);
    printf("%d out\n", interval_out);

    // Ending code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1072.hs
-- https://judge.beecrowd.com/pt/problems/view/1072
-- Intervalo 2

-- Importing useful libraries
import Control.Monad (replicateM) -- To iterate after the user enters the number of iterations

-- Function to get sequencial values
getValue :: IO Int
getValue = do
    input <- getLine
    let value = read input :: Int
    return value

-- Main
main :: IO ()
main = do        
    -- Requesting number of subsequent inputs
    usr_input <- readLn :: IO Int
    values <- replicateM usr_input getValue
    
    -- iterating
    let interval_in = length (filter (\x -> x >= 10 && x <= 20) values)
    let interval_out = length (filter (\x -> x < 10 || x > 20) values)

    -- Showing results
    putStrLn $ show interval_in ++ " in"
    putStrLn $ show interval_out ++ " out"

-- By Fernando Serra
-- https://github.com/fernandosserra